### PR TITLE
health-insurance - refactor

### PR DESCRIPTION
**Pull request by qdd1**
# \<Description\>

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by dabbeg — APPROVED** [review]
LGTM

**Review by essgje — APPROVED** [review]
lftm

File: apps/api/src/app/environments/environment.prod.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default {
   },
   healthInsurance: {
     wsdlUrl: process.env.HEALTH_INSURANCE_XROAD_WSDLURL,
-    baseUrl: process.env.XROAD_BASE_PATH_WITH_ENV,
+    baseUrl: process.env.XROAD_BASE_PATH,
```
Comment (baering) [inline apps/api/src/app/environments/environment.prod.ts:34]: For anyone else wondering why this is being changed, [here's a relevant discussion](https://github.com/island-is/island.is/pull/3095#discussion_r586327870).

File: libs/health-insurance/src/lib/soap/soapClient.ts [hunk 2]
```diff
@@ -73,4 +65,19 @@ export class SoapClient {
...
+      logger.error(`Could not find variables Ids from ${variables}`)
+      throw new Error(`Could not find variables Ids from ${variables}`)
+    }
+
+    return {
+      xRoadInstance: varArr[0],
+      memberClass: varArr[1],
+      memberCode: varArr[2],
+      subSystemCode: varArr[3],
+    }
```
Comment (baering) [inline libs/health-insurance/src/lib/soap/soapClient.ts:81]: This is completely fine, just to point it out you could also do:

```typescript
const [xRoadInstance, memberClass, memberCode, subSystemCode] = varArr

return {
  xRoadInstance,
  memberClass,
  memberCode,
  subSystemCode,
}
```

**Pull request merged**